File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/1973.c
#include <stdio.h>

int main(){
    long long n;
    scanf("%lld", &n);
    long long v[1000010], i, soma=0, maxdir=0;
    for(i=0;i<n;i++){
        scanf("%lld", &v[i]);
        soma+=v[i];
    }
    
    i=0;

    while(1){
        long long c = i;
        if(v[i]%2==0) i--;
        else i++;
        if(v[c]>0){
            v[c]--;
            soma--;
        }
        if(i<0 || i>=n) break;
        if(i>maxdir) maxdir=i;
    }
    
    printf("%lld %lld\n", maxdir+1, soma);
    
    return 0;
}

File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/1973.py
n = int(input())
soma = 0
maxdir = 0

v = [int(i) for i in input().split()]

for i in range(n):
    soma += v[i]

i = 0

while True:
    c = i;
    if(v[i] % 2 == 0): i -= 1
    else: i += 1
    
    if(v[c] > 0):
        v[c] -= 1
        soma -= 1
    
    if(i < 0 or i >= n): break
    if(i > maxdir): maxdir = i
    
print(str(maxdir+1)+' '+str(soma))